fineract-provider/src/test/java/org/apache/fineract/portfolio/savings/service/FixedDepositAccountInterestCalculationServiceImplTest.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.portfolio.savings.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.math.BigDecimal;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class FixedDepositAccountInterestCalculationServiceImplTest {
+
+    @InjectMocks
+    FixedDepositAccountInterestCalculationServiceImpl service;
+
+    @Test
+    public void testCalculateInterestInternal1() {
+
+        // Calculate interest
+        double expectedInterest = 10455.0624; // Expected interest calculated based on provided values
+        double calculatedInterest = service.calculateInterestInternal(10000L, BigDecimal.valueOf(4.5), 12L, 6L).doubleValue();
+
+        // Verify the result
+        assertEquals(expectedInterest, calculatedInterest, 0.001);
```

Comment:
We should not have any delta...

Reply:
Hello, delta of .001 was given to provide a little room to adjust the code and not result in test case failing with slightest change in code since the calculated value is highly precise.
What's your suggestion?

Reply:
I dont think we need delta here. Just simply check whether they are equal or not...